cypress/integration/oauth/change-password.spec.js

```diff
@@ -0,0 +1,115 @@
+// limitations under the License.
+
+describe('OAuth change password', () => {
+  before(() => {
+    cy.dropAndSeedDatabase()
+  })
+
+  it('should display UI elements in place', () => {
+    const user = { user_id: 'admin', password: 'admin' }
+    cy.loginOAuth(user)
+    cy.visit(`${Cypress.config('oauthRootPath')}/update-password`)
+
+    cy.findByText('Change password', { selector: 'h1' }).should('be.visible')
+    cy.findByLabelText('Old password').should('be.visible')
+    cy.findByLabelText('New password').should('be.visible')
+    cy.findByLabelText('Confirm password').should('be.visible')
+    cy.findByLabelText('Revoke access')
+      .should('be.visible')
+      .should('have.attr', 'value', 'true')
+    cy.findWarningByLabelText('Revoke access')
+      .should('contain', 'This will revoke access from all logged in devices')
+      .and('be.visible')
+    cy.findByRole('button', { name: 'Change password' }).should('be.visible')
+    cy.findByRole('button', { name: 'Cancel' }).should('be.visible')
+    cy.title().should('eq', `Change password - ${Cypress.config('siteName')}`)
+  })
+
+  it('should validate before submitting an empty form', () => {
+    const user = {
+      ids: { user_id: 'test-user-id1' },
+      primary_email_address: 'test-user1@example.com',
+      password: 'ABCDefg123!',
+      password_confirm: 'ABCDefg123!',
+    }
+    cy.createUser(user)
+    cy.loginOAuth({ user_id: user.ids.user_id, password: user.password })
+    cy.visit(`${Cypress.config('oauthRootPath')}/update-password`)
+    cy.findByRole('button', { name: 'Change password' }).click()
+
+    cy.findErrorByLabelText('New password').as('newPasswordError')
+    cy.get('@newPasswordError')
+      .should('contain', 'New password')
+      .and('be.visible')
+    cy.get('@newPasswordError')
+      .should('contain', 'is required')
+      .and('be.visible')
+    cy.findErrorByLabelText('Confirm password').as('confirmPasswordError')
+    cy.get('@confirmPasswordError')
+      .should('contain', 'Confirm password')
+      .and('be.visible')
+    cy.get('@confirmPasswordError')
+      .should('contain', 'is required')
+      .and('be.visible')
+
+    cy.location('pathname').should('eq', `${Cypress.config('oauthRootPath')}/update-password`)
+  })
+
+  it('should successfully change password w/ revoking access', () => {
+    const newPassword = 'ABCDefg321!'
+    const user = {
+      ids: { user_id: 'test-user-id2' },
+      primary_email_address: 'test-user2@example.com',
+      password: 'ABCDefg123!',
+      password_confirm: 'ABCDefg123!',
+    }
+    cy.createUser(user)
+    cy.loginOAuth({ user_id: user.ids.user_id, password: user.password })
+    cy.visit(`${Cypress.config('oauthRootPath')}/update-password`)
+
+    cy.findByLabelText('Old password').type(user.password)
+    cy.findByLabelText('New password').type(newPassword)
+    cy.findByLabelText('Confirm password').type(`${newPassword}{enter}`)
+
```

Comment:
```suggestion
cy.findByLabelText('Revoke access').check()
```

To keep this working regardless of the default state.